### PR TITLE
Add retreat 2025 + event template

### PR DESCRIPTION
**Pull request by aebelenkiy**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,3 +14,5 @@ gem 'jekyll-sass-converter', '2.2.0'
 gem 'csv'
 gem 'base64'
 gem 'bigdecimal'
+gem 'tzinfo'
+gem 'tzinfo-data'
```
Comment (jessesquires) [inline Gemfile:18]: Why do we need these?

File: _config.yml [hunk 2]
```diff
@@ -62,3 +63,14 @@ author:
...
+# Site settings
+baseurl: ""  # leave empty if serving from root
+url: ""      # leave empty for local development
+
+# Asset settings
+defaults:
+  - scope:
+      path: "assets/images"
+    values:
+      image: true
```
Comment (jessesquires) [inline _config.yml:76]: Image handling is all setup. I'll comment inline below with fixes.

I think everything in `_config.yml` here can be reverted.

----

RE: `baseurl:` and `url:` 

Were you having issues previewing the site locally?

File: _data/navigation.yml [hunk 3]
```diff
@@ -4,5 +4,12 @@
 - title: about
   page: about.md
 
+- title: events
+  page: events.md
+
+- title: 2025 retreat
+  page: retreat.md
```
Comment (jessesquires) [inline _data/navigation.yml:11]: I think we should call this `retreat2025.md`

File: _data/retreat_presenters.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+- name: Eden Thomson
+  image: assets/images/presenters/eden-thomson.png
```
Comment (jessesquires) [inline _data/retreat_presenters.yml:2]: RE: image handling

This should change to `image: eden-thomson.png`

And you can put the image in `img/`.

File: _data/retreat_presenters.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    techniques, and risk assessment.
+  bio: >
+    Eden possesses extensive experience spanning nearly 18 years within the body 
+    piercing and suspension domain. His organization, Immersive Environments, has 
+    successfully hosted numerous international events, including the Australasian 
+    Suscon, the annual Immersive Environments Retreat and the 2025 Otautah Hook 
+    Suspension Symposium. Eden's unwavering dedication lies in the realm of body 
+    suspension education, where he passionately advocates for the dissemination 
+    of standardized testing protocols and procedures to ensure the safety and 
+    well-being of the broader community.
```
Comment (jessesquires) [inline _data/retreat_presenters.yml:17]: minor nit: missing newline at end of file 😊

File: _layouts/event-page.html [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+      {% endif %}
+    </div>
+  </header>
+
+  <div class="event-content">
+    {{ content }}
+  </div>
+</article>
+
+<style>
```
Comment (jessesquires) [inline _layouts/event-page.html:35]: We can remove all of this style and use Bootstrap classes and components.

Quite a lot to explain, but if you're ok with it, I can push some commits here.

-----

One small example: 

Instead of having a custom class for margins like this:

```css
.event-header {
     margin-bottom: 3rem;

   }
```

Bootstrap provides a bunch of utility classes, so you can write `<header class="mb-3">` instead.

Bootstrap also provides lots of button styles and everything else we'll need.

https://getbootstrap.com/docs/5.3/components/buttons/

File: assets/images/presenters/eden-thomson.png
Comment (jessesquires) [inline assets/images/presenters/eden-thomson.png:1]: This file is **massive**.

We should convert to jpg and resize so that the largest dimension is no more than 1000px.

Then I'll run all the images through [ImageOptim](https://imageoptim.com/mac) to further compress them.

File: retreat.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
+---
+layout: event-page
+title: Retreat
+event_date: 2025
+location: Studio Friction, Denver
+registration_status: open
+featured_image: assets/images/studio-friction.jpg
```
Comment (jessesquires) [inline retreat.md:7]: RE: image handling

This should change to `image: studio-friction.jpg` and we can place the file in `img/`

File: _data/retreat_presenters.yml
Comment (jessesquires) [inline _data/retreat_presenters.yml:1]: Probably better to put this in `_data/retreat2025/presenters.yml`

Then you access it via `{{ site.data.retreat2025.presenters }}`

File: retreat.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+- New hooks available for purchase (advance sterilization required)
+- Personal hooks must be in unopened sterile packaging
+
+#### COVID Policy
+- Masks encouraged during travel
+- Rapid tests available if needed
+- Hand sanitizer provided throughout venue
+- Those with symptoms may be required to mask
+
+{% if site.registration_open %}
```
Comment (jessesquires) [inline retreat.md:98]: Not sure if we need a site-wide variable for this... ?

**Review by jessesquires — CHANGES_REQUESTED** [review]

File: _layouts/event-page.html [hunk 9]
```diff
@@ -0,0 +1,137 @@
+---
+layout: default
+---
+
+<article class="event-page">
+  <header class="event-header">
+    {% if page.featured_image %}
+    <div class="featured-image">
+      <img src="{{ page.featured_image }}" alt="{{ page.title }}">
```
Comment (jessesquires) [inline _layouts/event-page.html:9]: RE: image handling

There's a helper include for this already:

```
{% include image.html
file="photo.jpg"
alt="alt text"
%}
```

That's for main page in-line images. So that may or may not work for this use case.

Otherwise, for handling image `src`, we need to use this:

```
{% link {{ page.featured_image | img_url }} %}
```

Explanation:

- `{% link %}` validates that the file exists. If it's missing, `jekyll build` will fail. This will catch errors.
- `| img_url` forms the correct path to the file without having to hard-code paths

Basically, Jekyll has a bunch of built-in functionality to prevent missing/broken links so we should always avoid using raw paths.

File: _layouts/event-page.html
Comment (jessesquires) [inline _layouts/event-page.html:1]: Minor nit: I'd just call this `event.html`.

We know it's a page. 😊

File: Gemfile (hunk 1 above)
Comment (aebelenkiy) [reply Gemfile:18]: I was getting an error running builder without them for some reason

File: _layouts/event-page.html (hunk 9 above)
Comment (jessesquires) [reply _layouts/event-page.html:9]: For `img/` subdirectories:

Suppose we have `img/retreat2025/`

`{% link {{ page.featured_image | img_url: "retreat2025" }} %}`

File: _layouts/event.html [hunk 10]
```diff
@@ -0,0 +1,74 @@
+---
+layout: default
+---
+
+{% assign event_id = page.event_id %}
+{% assign event = site.data[event_id] %}
+{% assign details = event.details %}
+{% assign presenters = event.presenters %}
```
Comment (jessesquires) [inline _layouts/event.html:8]: @aebelenkiy ok so I made a bunch of progress on the template.

This is still not finished, but I think this should give you a good idea of how to do things.

The page now has an `event_id:` in the front matter, which we then use to map to `_data/retreat2025/...` where we have:

- `_data/retreat2025/details.yml`
- `_data/retreat2025/presenters.yml`

File: retreats/2025.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
+---
+layout: event
+title: Suspension Retreat 2025
+image: venues/studio_friction.jpg
+event_id: retreat2025
```
Comment (jessesquires) [inline retreats/2025.md:5]: Continuing from comment above:

So this maps to `_data/retreat2025/...`.

This allows use to have structured data in yaml files in `_data/retreat2025/` and makes all of this reusable and easier to work with.

File: retreats/2025.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+---
+layout: event
+title: Suspension Retreat 2025
+image: venues/studio_friction.jpg
```
Comment (jessesquires) [inline retreats/2025.md:4]: this is a site-level attribute.

this is basically a page's "featured image" --- aka, the image used for link previews on social media, etc.

You can see how this works here:

https://github.com/DenverSuspensionCollective/DenverSuspensionCollective.github.io/blob/main/_includes/head.html#L17

File: _data/retreat2025/presenters.yml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    successfully hosted numerous international events, including the Australasian
+    Suscon, the annual Immersive Environments Retreat and the 2025 Otautah Hook
+    Suspension Symposium. Eden's unwavering dedication lies in the realm of body
+    suspension education, where he passionately advocates for the dissemination
+    of standardized testing protocols and procedures to ensure the safety and
+    well-being of the broader community.
+
+- name: Sasha Belenkiy
+  image: sasha.jpg
+  presentation_title: How to be a Dumb Bitch
```
Comment (jessesquires) [inline _data/retreat2025/presenters.yml:21]: just some fake data for now to test rendering, etc. 😉

File: retreats/2025.md [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+event_id: retreat2025
+---
+
+<br><br><br><br><br><br><br><br>
```
Comment (jessesquires) [inline retreats/2025.md:8]: Everything below here is "ToDo" to be included in the yaml + template.

If `_data/retreat2025/details.yml` starts to get too big, we can create a new file.

Maybe what's below here should go in `_data/retreat2025/important_info.yml`
